instrumentation/github.com/gorilla/mux/otelmux/mux.go

```diff
@@ -47,31 +47,39 @@ func Middleware(service string, opts ...Option) mux.MiddlewareFunc {
 	}
 	tracer := cfg.TracerProvider.Tracer(
 		tracerName,
-		oteltrace.WithInstrumentationVersion(SemVersion()),
+		trace.WithInstrumentationVersion(SemVersion()),
 	)
 	if cfg.Propagators == nil {
 		cfg.Propagators = otel.GetTextMapPropagator()
 	}
 	if cfg.spanNameFormatter == nil {
 		cfg.spanNameFormatter = defaultSpanNameFunc
 	}
+	if cfg.PublicEndpointFn == nil {
+		cfg.PublicEndpointFn = func(*http.Request) bool { return false }
```

Comment:
Setting this to a non-nil value and then checking below if this value is nil doesn't make sense. Why not just leave this as `nil` and have the guard below prevent the nil deref?